pandas/core/arrays/interval.py

```diff
@@ -777,6 +777,17 @@ def take(self, indices, allow_fill=False, fill_value=None, axis=None,
 
         return self._shallow_copy(left_take, right_take)
 
+    def where(self, cond, other):
+        if is_scalar(other) and isna(other):
+            lother = rother = other
+        else:
```

Comment:
Can you make this an `elif` that checks that `other` is interval-like (something like `isinstance(other, Interval) or is_interval_dtype(other)`), then have an `else` clause that raises a `ValueError` saying `other` must be interval-like?  

As written I think this would raise a somewhat unclear `AttributeError` in `self._check_closed_matches` since it assumes `other.closed` exists.